Makefile

```diff
@@ -34,106 +26,177 @@ WEBHOOK_ROOT ?= $(MANIFEST_ROOT)/webhook
-	cd $(TOOLS_DIR); go build -tags=tools -o $(BIN_DIR)/mockgen github.com/golang/mock/mockgen
+$(TOOLBIN)/mockgen:
+	GOBIN=$(TOOLBIN) GO111MODULE=on go get github.com/golang/mock/mockgen@v1.3.1
 
-.PHONY: $(CONVERSION_GEN)
-$(CONVERSION_GEN): $(TOOLS_DIR)/go.mod
-	cd $(TOOLS_DIR); go build -tags=tools -o $(BIN_DIR)/conversion-gen k8s.io/code-generator/cmd/conversion-gen
+$(TOOLBIN)/conversion-gen:
+	GOBIN=$(TOOLBIN) GO111MODULE=on go get k8s.io/code-generator/cmd/conversion-gen@v0.17.0
 
-.PHONY: $(KUBEBUILDER)
-$(KUBEBUILDER): $(TOOLS_DIR)/go.mod
+$(TOOLBIN)/kubebuilder:
 	cd $(TOOLS_DIR); ./install_kubebuilder.sh
 
-.PHONY: $(KUSTOMIZE)
-$(KUSTOMIZE): $(TOOLS_DIR)/go.mod
+$(TOOLBIN)/kustomize:
 	cd $(TOOLS_DIR); ./install_kustomize.sh
 
-## --------------------------------------
-## Linting
-## --------------------------------------
+$(TOOLBIN)/kind:
+	GOBIN=$(TOOLBIN) GO111MODULE=on go get sigs.k8s.io/kind@v0.6.0
 
+$(TOOLBIN)/addlicense:
+	GOBIN=$(TOOLBIN) GO111MODULE=on go get github.com/google/addlicense
 
-all: test manager
+$(TOOLBIN)/misspell:
+	GOBIN=$(TOOLBIN) GO111MODULE=on go get github.com/client9/misspell/cmd/misspell@v0.3.4
+
+.PHONY: install-tools
+install-tools: \
+	$(TOOLBIN)/controller-gen \
+	$(TOOLBIN)/golangci-lint \
+	$(TOOLBIN)/mockgen \
+	$(TOOLBIN)/conversion-gen \
+	$(TOOLBIN)/kubebuilder \
+	$(TOOLBIN)/kustomize \
+	$(TOOLBIN)/addlicense \
+	$(TOOLBIN)/misspell \
+	$(TOOLBIN)/kind
+
+## --------------------------------------
+## Tests
+## --------------------------------------
 
 # Run tests
-test: generate fmt vet manifests
+.PHONY: test
+test:
 	go test -v ./api/... ./controllers/... -coverprofile $(COVER_FILE)
 
+# Run e2e-tests
+.PHONY: e2e-test
+e2e-test: generate fmt vet manifests
+	BIN=$(TOOLBIN) ./e2e/test_e2e.sh
+
+## --------------------------------------
+## Build and run
+## --------------------------------------
+
 # Build manager binary
-manager: generate fmt vet
+bin/manager: main.go generate fmt vet manifests
 	go build -o bin/manager main.go
 
 # Run against the configured Kubernetes cluster in ~/.kube/config
+.PHONY: run
 run: generate fmt vet manifests
 	go run ./main.go
 
 # Debug using the configured Kubernetes cluster in ~/.kube/config
-debug: generate fmt vet
+.PHONY: debug
+debug: generate fmt vet manifests
 	dlv debug ./main.go
 
-# Run go fmt against code
+## --------------------------------------
+## Code maintenance
+## --------------------------------------
+
+.PHONY: fmt
 fmt:
 	go fmt ./api/... ./controllers/...
 
-# Run go vet against code
+.PHONY: vet
 vet:
 	go vet ./api/... ./controllers/...
 
+.PHONY: fix
+fix:
+	go fix ./api/... ./controllers/...
+
+.PHONY: license
+license: $(TOOLBIN)/addlicense
+	$(TOOLBIN)/addlicense  -y $(shell date +"%Y") -c "The Kubernetes Authors." -f LICENSE_TEMPLATE .
+
+.PHONY: tidy
+tidy:
+	go mod tidy
+
+.PHONY: lint
+lint: $(TOOLBIN)/golangci-lint
+	$(TOOLBIN)/golangci-lint run ./api/... ./controllers/...
```

Comment:
I would prefer running glangci-lint on all packages `./...`, as `main.go` is missing from `./api/... ./controllers/...`

Reply:
ack will do.